Description: If not started with the session, we should always raise
Bug: bugs.launchpad.net/bugs/503052

=== modified file 'src/empathy-status-icon.c'
Index: empathy-2.31.5.1/src/empathy-status-icon.c
===================================================================
--- empathy-2.31.5.1.orig/src/empathy-status-icon.c	2010-07-13 17:06:09.000000000 +0200
+++ empathy-2.31.5.1/src/empathy-status-icon.c	2010-07-13 17:07:39.000000000 +0200
@@ -849,6 +849,9 @@
 #ifdef HAVE_LIBINDICATE
 	gboolean               use_libindicate;
 #endif
+	const gchar *desktop_autostart_id;
+
+	desktop_autostart_id = g_getenv ("DESKTOP_AUTOSTART_ID");
 
 	g_return_val_if_fail (GTK_IS_WINDOW (window), NULL);
 
@@ -884,8 +887,16 @@
 			  icon);
 
 	if (!hide_contact_list) {
-		should_hide = g_settings_get_boolean (priv->gsettings_ui,
-			EMPATHY_PREFS_UI_MAIN_WINDOW_HIDDEN);
+		/* If not started with the session, we should raise the window.
+	   https://bugs.launchpad.net/ubuntu/+source/empathy/+bug/503052 */
+		if (desktop_autostart_id == NULL) {
+			DEBUG ("Not started with session, showing contact list");
+			should_hide = FALSE;
+		} else {
+			DEBUG ("Auto-started with session, so honor the previous known state");
+			should_hide = g_settings_get_boolean (priv->gsettings_ui,
+					EMPATHY_PREFS_UI_MAIN_WINDOW_HIDDEN);
+		}
 	} else {
 		should_hide = TRUE;
 	}
